### PR TITLE
added deprecation for Product Pricing field

### PR DESCRIPTION
**Pull request by koprivajakub**
#### Changelog notes 

```
* Added deprecation for Product Pricing field
```

#### Check during review

- [ ] JSON example extended.
  - [ ] New properties are added to the correct place in the JSON.
- [ ] New properties in the table are added to the correct place.
- [ ] Correct formatting:
  - [ ] Spacing is consistent between titles, sections, tables, ...
  - [ ] Correct JSON format - indentation.
- [ ] DateTime properties should always be defined in ISO format.

### CODE REVIEW

File: distributor-api-v1/operations.md [hunk 1]
```diff
@@ -517,10 +506,36 @@ If the hotel does not use any payment gateway, the value is null. If it does, th
...
+##### Pricing data discriminator
+
+* `Absolute` - Data specific to absolutely priced product are represented by [Multi-currency amount](./operations.md#multi-currency-amount).
+* `Relative` - Data specific to relatively priced product are represented by [Relative price data value](./operations.md#relative-price-data-value).
+
+##### Relative price data value
+
+| Property | Type | Contract | Description |
+| :--- | :--- | :--- | :--- |
+| `ProductIds` | array of string | required | Product ids used to calculate the price of the relatively priced product. |
```
Comment (JakubSaksa) [inline distributor-api-v1/operations.md:529]: Nit: Maybe `array of strings`? `Array of string` sounds weird to me.

Comment (JakubSaksa) [reply distributor-api-v1/operations.md:529]: Nit: I would go with product `IDs` instead of `ids`.

**Review by JakubSaksa — APPROVED** [review]

File: distributor-api-v1/operations.md [hunk 2]
```diff
@@ -98,42 +98,69 @@ Preferred initial call used to obtain all static data about distributor configur
...
+                                    "GrossValue": 108.31,
+                                    "NetValue": 98.46,
+                                    "TaxValues": [
+                                        {
+                                            "TaxRateCode": "CZ-L",
+                                            "Value": 9.85
+                                        }
+                                    ]
+                                },
                             "EUR": {
```
Comment (KaliCZ) [inline distributor-api-v1/operations.md:123]: Got some weird indents here.

File: distributor-api-v1/operations.md [hunk 3]
```diff
@@ -517,10 +506,36 @@ If the hotel does not use any payment gateway, the value is null. If it does, th
...
+* `Relative` - Data specific to relatively priced product are represented by [Relative price data value](./operations.md#relative-price-data-value).
+
+##### Relative price data value
+
+| Property | Type | Contract | Description |
+| :--- | :--- | :--- | :--- |
+| `ProductIds` | array of string | required | Product ids used to calculate the price of the relatively priced product. |
+| `Multiplier` | number | required | Percentage of the relative price. |
+| `Target` | [Relative price target](./operations.md#relative-price-target) | required | Target defining whether the price of a product should be calculated from gross or net value of dependant products. |
+| `TaxRateCodes` | array of string | required | Tax rate codes that should be applied to the price in order to calculate gross price of the product. |
```
Comment (KaliCZ) [inline distributor-api-v1/operations.md:532]: I wouldn't say `to calculate gross price of product.` In gross-priced environments, gross comes first and then taxes are deducted from the gross in order to calculate net. 

Maybe just `in order to calculate the taxes of the product`?

File: distributor-api-v1/operations.md [hunk 4]
```diff
@@ -517,10 +506,36 @@ If the hotel does not use any payment gateway, the value is null. If it does, th
...
+
+##### Relative price target
+
+* `GrossValue` - The price of the product should be calculated from gross value of dependant products
+* `NetValue` - The price of the product should be calculated from net value of dependant products
```
Comment (KaliCZ) [inline distributor-api-v1/operations.md:537]:
```suggestion
* `GrossValue` - The price of the product should be calculated from gross value of dependant products
* `TaxValue` - The price of the product should be calculated from tax value of dependant products
* `NetValue` - The price of the product should be calculated from net value of dependant products
```

File: distributor-api-v1/operations.md (hunk 1 above)
Comment (koprivajakub) [reply distributor-api-v1/operations.md:529]: I am following the same standard as Connector here. So vice versa, I did update the occurance of `array of strings` to `array of string` everywhere to clean it and use only one version.

`ids` renamed to `Unique identifiers` again, globally in the entire file.

**Review by JakubSaksa — APPROVED** [review]

File: distributor-api-v1/operations.md [hunk 5]
```diff
@@ -517,10 +572,37 @@ If the hotel does not use any payment gateway, the value is null. If it does, th
...
+* `Absolute` - Data specific to absolutely priced product are represented by [Multi-currency amount](./operations.md#multi-currency-amount).
+* `Relative` - Data specific to relatively priced product are represented by [Relative price data value](./operations.md#relative-price-data-value).
+
+##### Relative price data value
+
+| Property | Type | Contract | Description |
+| :--- | :--- | :--- | :--- |
+| `ProductIds` | array of string | required | Unique identifiers of [product](./operations.md#product)s used to calculate the price of the relatively priced product. |
+| `Multiplier` | number | required | Percentage of the relative price. |
+| `Target` | [Relative price target](./operations.md#relative-price-target) | required | Target defining whether the price of a product should be calculated from gross or net value of dependant products. |
```
Comment (KaliCZ) [inline distributor-api-v1/operations.md:597]:
```suggestion
| `Target` | [Relative price target](./operations.md#relative-price-target) | required | Target defining whether the price of a product should be calculated by multiplying gross value, tax value or net value of dependant products. |
```

**Review by KaliCZ — APPROVED** [review]

**Pull request merged**